rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/history/HistoryRepositorySpec.scala

```diff
@@ -0,0 +1,134 @@
+package coop.rchain.rspace.nextgenrspace.history
+
+import coop.rchain.rspace.{
+  util,
+  Blake2b256Hash,
+  DeleteContinuations,
+  DeleteData,
+  DeleteJoins,
+  HotStoreAction,
+  InsertContinuations,
+  InsertData,
+  InsertJoins
+}
+import monix.eval.Task
+import org.scalatest.{FlatSpec, Matchers, OptionValues}
+
+import scala.concurrent.duration._
+import monix.execution.Scheduler.Implicits.global
+import coop.rchain.rspace.internal.{Datum, WaitingContinuation}
+import coop.rchain.rspace.nextgenrspace.history.TestData.randomBlake
+import coop.rchain.rspace.trace.{Consume, Produce}
+
+import scala.collection.concurrent.TrieMap
+import scala.util.Random
+import cats.implicits._
+import scodec.Codec
+
+class HistoryRepositorySpec
+    extends FlatSpec
+    with Matchers
+    with OptionValues
+    with InMemoryHistoryRepositoryTestBase {
+
+  type TestHistoryRepository = HistoryRepository[Task, String, String, String, String]
+
+  "HistoryRepository" should "process insert one datum" in withEmptyRepository { repo =>
+    val testDatum = datum(1)
+    val data      = InsertData[String, String](testChannelDataPrefix, testDatum :: Nil)
+    for {
+      nextRepo <- repo.process(data :: Nil)
+      data     <- nextRepo.getData(testChannelDataPrefix)
+      fetched  = data.head
+      _        = fetched shouldBe testDatum
+    } yield ()
+  }
+
+  val testChannelDataPrefix          = "channel-data"
+  val testChannelJoinsPrefix         = "channel-joins"
+  val testChannelContinuationsPrefix = "channel-continuations"
+
+  it should "process insert and delete of thirty mixed elements" in withEmptyRepository { repo =>
+    val data  = (0 to 10).map(insertDatum).toVector
+    val joins = (0 to 10).map(insertJoin).toVector
+    val conts = (0 to 10)
+      .map(insertContinuation)
+      .toVector
+    val elems: Vector[HotStoreAction] = Random.shuffle(data ++ joins ++ conts)
+
+    val dataDelete                             = data.map(d => DeleteData[String](d.channel))
+    val joinsDelete                            = joins.map(j => DeleteJoins[String](j.channel))
+    val contsDelete                            = conts.map(c => DeleteContinuations[String](c.channels))
+    val deleteElements: Vector[HotStoreAction] = dataDelete ++ joinsDelete ++ contsDelete
```

Comment:
I would change this test into a proper generative test

Reply:
it already is. this is a simple unit.

Reply:
technically it generates some random data but it doesn't use a proper Arbitrary instance